pkg/scale/decode.go

```diff
@@ -108,6 +111,20 @@ type decodeState struct {
 }
 
 func (ds *decodeState) unmarshal(dstv reflect.Value) (err error) {
+	// Handle BTreeMap type separately for the following reasons:
+	// 1. BTreeMap is a generic type, so we can't use the normal type switch
+	// 2. We cannot use BTreeCodec because we are comparing the type of the dstv.Interface() in the type switch
+	if isBTree(dstv.Type()) {
```

Comment:
`isBtree` also checks for `BTree` and `BTreeMap` right? so is the switch case on line `150` needed?

Reply:
Oh yeah, I meant to remove that. Thank you!